modules/annotations/shared/src/main/scala/io/circe/derivation/annotations/Configuration.scala

```diff
@@ -18,12 +18,20 @@ sealed trait Configuration {
    */
   def transformMemberNames: String => String
 
+  /** Transforms the value of any constructor names in the JSON, allowing,
+   *  for example, formatting or case changes
+   */
+  def transformConstructorNames: String => String
+
   def useDefaults: Boolean
 
   def discriminator: Option[String]
 
   protected def getA(transformMemberNames: String => String): Config
 
+  // TODO: Naming???
+  protected def getC(transformConstructorNames: String => String): Config
```

Comment:
I wasn't sure if the existing `getA` meant something that I should try to mimic here. I chose `C` for constructor 😄

Reply:
That was @aparo's name. I don't personally feel strongly about it, and given that they're protected and on a sealed trait I don't think we need to worry too much about the decision.